File: ePozoriste.Model/Termin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ePozoriste.Model
{
    public class Termin
    {
        public int TerminId { get; set; }
        public bool Premijera { get; set; }
        public bool Predpremijera { get; set; }
        public int CijenaKarte { get; set; }
        public DateTime DatumVrijemeOdrzavanja { get; set; }

        public int? SalaId { get; set; }
        public virtual Sala? Sala { get; set; }
        public int? PredstavaId { get; set; }
        public virtual Predstava? Predstava { get; set; }
        
    }
}

File: epozoriste_mobile/lib/models/termin.dart
import 'package:epozoriste_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'termin.g.dart';

@JsonSerializable()
class Termin {
  int terminId;
  bool premijera;
  bool predpremijera;
  int cijenaKarte;
  DateTime datumOdrzavanja;
  String vrijemeOdrzavanja;
  int? salaId;
  Sala sala;
  int? predstavaId;
  Predstava predstava;

  Termin(
      {required this.terminId,
      required this.premijera,
      required this.predpremijera,
      required this.cijenaKarte,
      required this.datumOdrzavanja,
      required this.vrijemeOdrzavanja,
      this.salaId,
      required this.sala,
      this.predstavaId,
      required this.predstava});

  factory Termin.fromJson(Map<String, dynamic> json) => _$TerminFromJson(json);
  Map<String, dynamic> toJson() => _$TerminToJson(this);
}
